apps/server/src/api-integration/integration.controller.ts

```diff
@@ -54,18 +54,15 @@ const actionHandlers: Record<string, ActionHandler> = {
       if (typeof property !== 'string' || value === undefined) {
         throw new Error('Invalid property or value');
       }
-
       // parseProperty is async because of the data lock
-      parseProperty(property, value).then((newObjectProperty) => {
-        const key = Object.keys(newObjectProperty)[0] as keyof OntimeEvent;
-        shouldThrottle = willCauseRegeneration(key) || shouldThrottle;
-
-        if (patchEvent.custom && newObjectProperty.custom) {
-          Object.assign(patchEvent.custom, newObjectProperty.custom);
-        } else {
-          Object.assign(patchEvent, newObjectProperty);
-        }
-      });
+      const newObjectProperty = parseProperty(property, value);
+      const key = Object.keys(newObjectProperty)[0] as keyof OntimeEvent;
+      shouldThrottle = willCauseRegeneration(key) || shouldThrottle;
+      if (patchEvent.custom && newObjectProperty.custom) {
+        Object.assign(patchEvent.custom, newObjectProperty.custom);
+      } else {
+        Object.assign(patchEvent, newObjectProperty);
+      }
     });
 
     if (shouldThrottle) {
```

Comment:
suggestion: i dont think the consumer needs to know about throttling. we can simplify like

```ts
    if (shouldThrottle) {
      throttledUpdateEvent(patchEvent)
    } else {
      updateEvent(patchEvent);
    }
    return { payload: 'success' };
```

Reply:
I would like to keep it, knowing that the API is limited is a good thing and a good user might even back off a bit